From 21217f5620bd4c1e03baa681acb08c8ac147f4ba Mon Sep 17 00:00:00 2001
From: Orlando Vazquez <ovazquez@gmail.com>
Date: Sat, 22 Aug 2009 11:32:16 -0700
Subject: [PATCH] Move the context menu and its logic from World to Hud


diff --git a/scripts/hud.py b/scripts/hud.py
index e4ea39f..9bc088a 100644
--- a/scripts/hud.py
+++ b/scripts/hud.py
@@ -18,6 +18,7 @@
 import shutil, fife, pychan
 from pychan.tools import callbackWithArguments as cbwa
 from scripts.parpgfilebrowser import PARPGFileBrowser
+from scripts.context_menu import ContextMenu
 import weakref
 
 class Hud(object):
@@ -41,6 +42,7 @@ class Hud(object):
         self.menu_displayed = False
         self.initializeHud()
         self.initializeMainMenu()
+        self.initializeContextMenu()
         self.initializeOptionsMenu()
         self.initializeHelpMenu()
         self.initializeEvents()
@@ -126,7 +128,22 @@ class Hud(object):
         """Hide the HUD.
            @return: None"""
         self.hud.hide()
-        
+
+    def initializeContextMenu(self):
+        """Initialize the Context Menu
+           @return: None"""
+        self.context_menu = ContextMenu (self.engine, [], (0,0))
+        self.context_menu.hide()
+
+    def showContextMenu(self, data, pos):
+        """Display the Context Menu with data at pos
+           @type data: list
+           @param data: data to pass to context menu
+           @type pos: tuple
+           @param pos: tuple of x and y coordinates
+           @return: None"""
+        self.context_menu = ContextMenu(self.engine, data, pos)
+
     def initializeMainMenu(self):
         """Initalize the main menu.
            @return: None"""
diff --git a/scripts/world.py b/scripts/world.py
index 1bffc7d..95eaf3e 100644
--- a/scripts/world.py
+++ b/scripts/world.py
@@ -23,7 +23,6 @@ from scripts.common.eventlistenerbase import EventListenerBase
 from settings import Setting
 from scripts import inventory, hud
 from scripts.popups import *
-from scripts.context_menu import ContextMenu
 from pychan.tools import callbackWithArguments as cbwa
 from map import Map
 
@@ -70,8 +69,6 @@ class World(EventListenerBase):
         self.inventory.inventory.mapEvents(self.inventory.events_to_map)
         self.refreshReadyImages()
         
-        self.context_menu = ContextMenu (self.engine, [], (0,0))
-        self.context_menu.hide()
         self.boxOpen = False
         self.boxCreated = False
         
@@ -214,7 +211,7 @@ class World(EventListenerBase):
            @type evt: fife.event
            @param evt: The event that fife caught
            @return: None"""
-        self.context_menu.hide() # hide the context menu in case it is displayed
+        self.hud.context_menu.hide() # hide the context menu in case it is displayed
         scr_point = fife.ScreenPoint(evt.getX(), evt.getY())
         if(evt.getButton() == fife.MouseEvent.LEFT):
             self.data.handleMouseClick(self.getCoords(scr_point))      
@@ -232,12 +229,12 @@ class World(EventListenerBase):
             # take the menu items returned by the engine or show a default menu if no items    
             data = info or [["Walk", "Walk here", self.onWalk, self.getCoords(scr_point)]]
             # show the menu
-            self.context_menu = ContextMenu(self.engine, data, (scr_point.x,scr_point.y))
+            self.hud.showContextMenu(data, (scr_point.x, scr_point.y))
 
     def onWalk(self, click):
         """Callback sample for the context menu.
         """
-        self.context_menu.hide()
+        self.hud.context_menu.hide()
         self.data.gameState.PC.run(click)
 
     def mouseMoved(self, evt):
@@ -274,13 +271,6 @@ class World(EventListenerBase):
         location.setMapCoordinates(coord)
         return location
 
-    def clearMenu(self):
-        """ Hides the context menu. Just nice to have it as a function.
-            @return: None"""
-        if hasattr(self, "context_menu"):
-            self.context_menu.vbox.hide()
-            delattr(self, "context_menu")
-
     def createBoxGUI(self, title):
         """
         Creates a window to display the contents of a box
-- 
1.6.0.4

